cdap-test/src/main/java/co/cask/cdap/test/MetricsManager.java

```diff
@@ -61,6 +63,26 @@ public MetricsManager(MetricStore metricStore) {
   }
 
   /**
+   * Search the metric store and return the collection of metric names for the search query
+   * @param query
+   * @return Collection of metric names
+   * @throws Exception
+   */
+  public Collection<String> searchMetricNames(MetricSearchQuery query) throws Exception {
```

Comment:
where is this method used?

Reply:
its good to add this method along with the search tags, ill add an additional test to check the task level metric isn't present in the returned list metric names, so we don't have an unused method here.

Reply:
You should add a test with better coverage. If you only add a test that the task level metrics are not present, then the test may succeed because this method does not work, or because they are actually not present. So you should at least add one test case for the positive case.